plugins/chat/app/services/chat_message_destroyer.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ChatMessageDestroyer
+  def destroy_in_batches(chat_messages_query, batch_size: 200)
+    chat_messages_query.in_batches(of: batch_size).each do |relation|
+      destroyed_ids = relation.destroy_all.pluck(:id)
+      reset_last_read_message_id(destroyed_ids)
+      delete_flags(destroyed_ids)
+    end
+  end
+
+  private
+
+  def reset_last_read_message_id(ids)
+    UserChatChannelMembership.where(last_read_message_id: ids).update_all(
+      last_read_message_id: nil,
+    )
+  end
```

Comment:
```suggestion
  def destroy_in_batches(chat_messages_query, batch_size: 200)
    chat_messages_query.in_batches(of: batch_size).each do |relation|
      destroyed_ids = relation.destroy_all.pluck(:id)
      reset_last_read(destroyed_ids)
      delete_flags(destroyed_ids)
    end
  end

  private

  def reset_last_read(message_ids)
    UserChatChannelMembership.where(last_read_message_id: message_ids).update_all(
      last_read_message_id: nil,
    )
  end
```

Reply:
Suggestion this to make it more consistent with `delete_flags`. Careful it might not pass linting.